app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/BaseViewModel.kt

```diff
@@ -22,20 +24,43 @@ import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import org.koin.core.component.KoinComponent
 import org.koin.core.component.inject
 import timber.log.Timber
 
 /** Provides [LiveEvent]s for both navigation and external navigation and helper functionality to observe the [LiveData] from a [Fragment] */
 abstract class BaseViewModel : ViewModel(), KoinComponent {
     protected val dispatcherProvider: DispatcherProvider by inject()
+    private val toaster: Toaster by inject()
 
     /**
      * Helper to launch to IO thread quickly
      */
     fun launchIO(block: suspend CoroutineScope.() -> Unit): Job =
         viewModelScope.launch(dispatcherProvider.IO, block = block)
 
+    suspend fun runOnMain(block: suspend CoroutineScope.() -> Unit) =
+        withContext(dispatcherProvider.Main, block)
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Error handling
+    ///////////////////////////////////////////////////////////////////////////
+    suspend fun handleError(messageId: Int) {
+        runOnMain {
+            toaster.toast(messageId)
+        }
+    }
+
+    suspend inline fun <T : Any> Status<T>.handlingErrors(messageID: Int , onSuccess: (T) -> Unit): Status<T> {
```

Comment:
@BottleRocket-Colin nit: Refactor argument name to `messageId` for project-wide naming consistency